tutorials/snapcraft/snap-a-nodejs-app.md

```diff
@@ -0,0 +1,666 @@
+  - How to enable and debug confinement issues
+  - Basics on plugs and interfaces
+  - And more...
+
+### What you’ll need
+  - Any supported snap GNU/Linux distribution .
+  - Some very basic knowledge of command line use, know how to edit files.
+  - You should have followed the 2 following tutorials: “[basic snap usage]” and "[create your first snap]”
+
+Survey
+: How will you use this tutorial?
+ - Only read through it
+ - Read it and complete the exercises
+: What is your current level of experience?
+ - Novice
+ - Intermediate
+ - Proficient
+
+## Getting familiar with the project
+Duration: 1:00
+
+This project is a **nodejs** application, serving a web page on request. The backend code issue a REST request through through a web API, using **chucknorris-io** npm module.
+
+### Get the code
+
+Simply run:
+
+
+```bash
+git clone https://github.com/ubuntu/snap-tutorials-code
+cd snap-tutorials-code/build-a-nodejs-service/start
+```
+
+Your cloned directory should contain 2 files:
+  - a `main.js` source file, containing the code. Opening this one will show the entire application logic. Notice that we listen on port 80. We will need to point to that file and having node running it as a service.
+  - a `package.json`, which is a standard npm metadata file, defining application name, version and dependency (chucknorris-io npm module in that case).
+
+
+## Start snapping your project
+Duration: 5:00
+
+### Metadata
+
+Let’s scaffold our default `snapcraft.yaml` file via:
+
+
+```bash
+$ snapcraft init
+Created snap/snapcraft.yaml.
+Edit the file to your liking or run `snapcraft` to get started
+```
+
+And let’s edit the top metadata to match our project definition:
+
+
+```bash
+name: chuck-norris-webserver
+version: '1.0.0'
+summary: Chuck Norris quotation nodejs server
+description: |
+  This example shows how to build a nodejs web server. This enables us to
+  demonstrate how service works, basic confinement rules to access and listen
+  on the network, as well as the nodejs snapcraft plugin.
+grade: stable
+confinement: devmode
+```
+
+As usual, note that we start in devmode. Let’s get to the serious stuff and add our nodejs part!
+
+### Adding a nodejs part
+
+Our application is using nodejs for execution. Luckily, snapcraft has a **nodejs plugin** enabling your to snap software using this technology in a breeze!
+It’s as simple as defining:
+
+
+```bash
+parts:
+  webserver:
+    source: .
+    plugin: nodejs
+```
+
+And that’s all what you need! Awesomely easy, isn’t it?
+The nodejs plugin snapcraft will download and embed a local nodejs version, as well as any dependencies that are shipped in your package.json file (like chucknorris-io in our case).
+
+
+positive
+: What if I don’t have any package.json file?
+snapcraft help nodejs shows you specific plugin options that you can use in your snapcraft.yaml. One of them is named node-packages which is a list of npm packages you may want to download in your snap.
```

Comment:
"snapcraft help nodejs", "snapcraft.yaml", "node-packages" are writtent as `code` in the original.